express-notes-api/server.js

```diff
@@ -0,0 +1,127 @@
+import express from 'express';
+import * as fs from 'node:fs/promises';
+
+const app = express();
+
+const data = JSON.parse(await fs.readFile('data.json'));
+const notes = data.notes;
+
+app.get('/api/notes', (req, res) => {
+  console.log(req.method);
+  const array = [];
+  for (const id in notes) {
+    array.push(notes[id]);
+  }
+  res.status(200).json(array);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  console.log(req.method);
+  const id = Number(req.params.id);
+  let outcome = {};
+  outcome.error = `Cannot find note with id ${id}`;
+  let statusCode = 404;
+  if (id < 0) {
+    statusCode = 400;
+    outcome.error = 'id must be a positive integer';
+  } else {
+    const array = [];
```

Comment:
This whole block (lines 28 to 37) can be shortened to just a single line of code (plus error checking) if you use object access. Study up on objects!
Hint: `outcome = notes[id];`